src/rendezvous/transports/simpleHttpTransport.ts

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { UnstableValue } from 'matrix-events-sdk';
+
+import { logger } from '../../logger';
+import { sleep } from '../../utils';
+import { RendezvousFailureListener, RendezvousFailureReason } from '../cancellationReason';
+import { RendezvousTransport, RendezvousTransportDetails } from '../transport';
+import { MatrixClient } from '../../matrix';
+import { ClientPrefix } from '../../http-api';
+
+const TYPE = new UnstableValue("http.v1", "org.matrix.msc3886.http.v1");
+
+export interface MSC3886SimpleHttpRendezvousTransportDetails extends RendezvousTransportDetails {
+    uri: string;
+}
+
+/**
+ * Implementation of the unstable [MSC3886](https://github.com/matrix-org/matrix-spec-proposals/pull/3886)
+ * simple HTTP rendezvous protocol.
+ * Note that this is UNSTABLE and may have breaking changes without notice.
+ */
+export class MSC3886SimpleHttpRendezvousTransport implements RendezvousTransport {
+    ready = false;
+    cancelled = false;
```

Comment:
```suggestion
    public ready = false;
    public cancelled = false;
```

also, it's a bit of a smell if these are mutable externally - should they be `private` with only a getter accessible?